drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/widget/ConditionPopup.java

```
@@ -435,282 +434,293 @@ ListBox loadPatterns() {
+        final FormStylePopup pop = new FormStylePopup(GuidedDecisionTableConstants.INSTANCE.FactType());
+
+        pop.addAttribute(GuidedDecisionTableConstants.INSTANCE.ChooseExistingPatternToAddColumnTo(),
+                         pats);
+
+        pop.add(new ModalFooterChangePattern(getShowPatternOKCommand(pats,
+                                                                     pop),
+                                             () -> {
+                                                 pop.hide();
+                                                 showNewPatternDialog();
+                                             },
+                                             () -> {
+                                                 pop.hide();
+                                                 view.enableFooter(true);
+                                             }
+        ));
 
-        pop.addAttribute( GuidedDecisionTableConstants.INSTANCE.ChooseExistingPatternToAddColumnTo(),
-                          pats );
+        view.enableFooter(false);
+        pop.show();
+    }
 
-        pop.add( new ModalFooterChangePattern( new Command() {
-            @Override
-            public void execute() {
-                String[] val = pats.getValue( pats.getSelectedIndex() ).split( "\\s" );
-                editingPattern = model.getConditionPattern( val[ 1 ] );
+    Command getShowPatternOKCommand(final ListBox pats,
+                                    final FormStylePopup pop) {
+        return () -> {
+            String[] val = pats.getValue(pats.getSelectedIndex()).split("\\s");
+            final Optional<Pattern52> p = extractEditingPattern(val);
 
-                //Clear Field and Operator when pattern changes
-                editingCol.setFactField( null );
-                editingCol.setOperator( null );
+            //This should never be thrown; but let's fail fast to make debugging easier
```

@Rikkola I've also added a comment about ```IllegalStateException``` - it shouldn't ever be thrown as it will suggest something rather more serious has gone wrong; but I thought it better to fail fast than have some weird exception elsewhere.